sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp

```diff
@@ -53,127 +55,129 @@ namespace Azure { namespace Storage { namespace Blobs {
-    {
-      return AccessTier::P2;
-    }
-    if (access_tier == "P3")
-    {
-      return AccessTier::P3;
-    }
-    if (access_tier == "P4")
-    {
-      return AccessTier::P4;
-    }
-    if (access_tier == "P6")
-    {
-      return AccessTier::P6;
-    }
-    if (access_tier == "P10")
-    {
-      return AccessTier::P10;
-    }
-    if (access_tier == "P15")
-    {
-      return AccessTier::P15;
-    }
-    if (access_tier == "P20")
-    {
-      return AccessTier::P20;
-    }
-    if (access_tier == "P30")
-    {
-      return AccessTier::P30;
-    }
-    if (access_tier == "P40")
-    {
-      return AccessTier::P40;
-    }
-    if (access_tier == "P50")
-    {
-      return AccessTier::P50;
-    }
-    if (access_tier == "P60")
-    {
-      return AccessTier::P60;
-    }
-    if (access_tier == "P70")
-    {
-      return AccessTier::P70;
-    }
-    if (access_tier == "P80")
+    inline AccessTier AccessTierFromString(const std::string& access_tier)
     {
-      return AccessTier::P80;
-    }
-    if (access_tier == "Hot")
-    {
-      return AccessTier::Hot;
-    }
-    if (access_tier == "Cool")
-    {
-      return AccessTier::Cool;
-    }
-    if (access_tier == "Archive")
-    {
-      return AccessTier::Archive;
+      if (access_tier == "")
+      {
+        return AccessTier::Unknown;
+      }
+      if (access_tier == "P1")
+      {
+        return AccessTier::P1;
+      }
+      if (access_tier == "P2")
+      {
+        return AccessTier::P2;
+      }
+      if (access_tier == "P3")
+      {
+        return AccessTier::P3;
+      }
+      if (access_tier == "P4")
+      {
+        return AccessTier::P4;
+      }
+      if (access_tier == "P6")
+      {
+        return AccessTier::P6;
+      }
+      if (access_tier == "P10")
+      {
+        return AccessTier::P10;
+      }
+      if (access_tier == "P15")
+      {
+        return AccessTier::P15;
+      }
+      if (access_tier == "P20")
+      {
+        return AccessTier::P20;
+      }
+      if (access_tier == "P30")
+      {
+        return AccessTier::P30;
+      }
+      if (access_tier == "P40")
+      {
+        return AccessTier::P40;
+      }
+      if (access_tier == "P50")
+      {
+        return AccessTier::P50;
+      }
+      if (access_tier == "P60")
+      {
+        return AccessTier::P60;
+      }
+      if (access_tier == "P70")
+      {
+        return AccessTier::P70;
+      }
+      if (access_tier == "P80")
+      {
+        return AccessTier::P80;
+      }
+      if (access_tier == "Hot")
+      {
+        return AccessTier::Hot;
+      }
+      if (access_tier == "Cool")
+      {
+        return AccessTier::Cool;
+      }
+      if (access_tier == "Archive")
+      {
+        return AccessTier::Archive;
+      }
+      throw std::runtime_error("cannot convert " + access_tier + " to AccessTier");
     }
-    throw std::runtime_error("cannot convert " + access_tier + " to AccessTier");
-  }
+  } // namespace Details
```

Comment:
Can you restructure/reorder the types in the file so you don't need to keep interspersing so many `Details` namespace in the middle?

Reply:
it's not easy to do for the current auto generator.

anyway, this generator is supposed to be a temporary thing, and core sdk team is supposed to provide us with a full-fledged generator.